### PR TITLE
Create Issue In Database

### PR DESCRIPTION
**Pull request by macrael**
This PR adds pop support to PPP.

### CODE REVIEW

File: bin/soda [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# This file allows running soda from the top level directory, it correctly runs with the correct go env vars and in the correct directory.
+
+GOPATH=$PWD/server
+export GOPATH
```
Comment (tenshiemi) [inline bin/soda]: Can 5 and 6 be a oneliner? In our Makefile we just do `export GOPATH = $(CURDIR)/server`

File: server/src/dp3/cmd/webserver/main.go [hunk 2]
```diff
@@ -44,6 +48,16 @@ func main() {
 	}
 	zap.ReplaceGlobals(logger)
 
+	//DB connection
+	pop.AddLookupPaths(*config)
+	dbConnection, err := pop.Connect("development")
```
Comment (tenshiemi) [inline server/src/dp3/cmd/webserver/main.go]: Should this be configurable? (e.g. pop.Connect("production"))

File: bin/soda (hunk 1 above)
Comment (macrael) [reply bin/soda]: shellcheck doesn't like that because it masks the return value of the first line.

File: bin/soda [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+GOBIN=$GOPATH/bin
```
Comment (macrael) [inline bin/soda]: what is GOBIN?

File: README.md [hunk 4]
```diff
@@ -57,17 +57,21 @@ There are a few handy targets in the Makefile to help you interact with the dev
...
 
-If you need to change the datata base schema, you'll need to write a migration.
+If you need to change the database schema, you'll need to write a migration.
 
 Creating a migration:
 
-_*TBD* It looks like we'll be using [Pop](https://github.com/markbates/pop), this section will need to be updated when we set it up._
+Use soda (a part of [pop](https://github.com/markbates/pop/)) to generate migrations. In order to make using soda easy, a wrapper is in `/bin/soda` that sets the go environment and working directory correctly.
+
+If you are generating a new model, use `/bin/soda generate model model-name column-name:type column-name:type ...`. id, created_at and updated_at are all created automatically.
```
Comment (macrael) [inline README.md]: ./

File: server/src/dp3/cmd/webserver/main.go [hunk 5]
```diff
@@ -25,8 +28,9 @@ func requestLogger(h http.Handler) http.Handler {
 
 func main() {
 
-	entry := flag.String("entry", "../client/build/index.html", "the entrypoint to serve.")
-	build := flag.String("build", "../client/build", "the directory of the built React app.")
+	entry := flag.String("entry", "client/build/index.html", "the entrypoint to serve.")
+	build := flag.String("build", "client/build", "the directory to serve static files from.")
+	config := flag.String("config-dir", "server/src/dp3/config", "The location of server config files")
```
Comment (macrael) [inline server/src/dp3/cmd/webserver/main.go]: maybe a less greedy name

File: server/src/dp3/cmd/webserver/main.go (hunk 2 above)
Comment (macrael) [reply server/src/dp3/cmd/webserver/main.go]: lets add a flag

File: server/src/dp3/migrations/schema.sql [hunk 6]
```diff
@@ -0,0 +1,80 @@
+--
```
Comment (macrael) [inline server/src/dp3/migrations/schema.sql]: lets ignore this for now.

File: bin/soda (hunk 3 above)
Comment (macrael) [reply bin/soda]: by default it's go path + /bin

File: server/src/dp3/cmd/webserver/main.go [hunk 7]
```diff
@@ -31,6 +31,7 @@ func main() {
 	entry := flag.String("entry", "client/build/index.html", "the entrypoint to serve.")
 	build := flag.String("build", "client/build", "the directory to serve static files from.")
 	config := flag.String("config-dir", "server/src/dp3/config", "The location of server config files")
+	env := flag.String("env", "dev", "The environment to run in, configures the database, presenetly.")
```
Comment (kilbergr) [inline server/src/dp3/cmd/webserver/main.go]: presently.

Comment (kilbergr) [inline server/src/dp3/cmd/webserver/main.go]: Should this include "test" env?

File: server/src/dp3/pkg/api/api_test.go [hunk 8]
```diff
@@ -48,5 +54,22 @@ func TestIndexIssuesHandler(t *testing.T) {
 	if resp.StatusCode != 200 {
 		t.Errorf("Returned status code: %d", resp.StatusCode)
 	}
```
Comment (kilbergr) [inline server/src/dp3/pkg/api/api_test.go]: I'd still like a check in here of whether there's a new issue created. I think that's an appropriate test for this feature. Please lmk if you disagree.

Comment (macrael) [reply server/src/dp3/pkg/api/api_test.go]: You mean by checking in the database?

Comment (kilbergr) [reply server/src/dp3/pkg/api/api_test.go]: Yes

**Pull request merged**